### PR TITLE
OCPBUGS-28647: tuned: operand: deferred updates prepare: cleanups from #1019

### PR DESCRIPTION
**Pull request by ffromani**
The PR #1019 includes a fair bit of cleanup and preparation code which may merge independently because these improvements are [liked anyway](https://github.com/openshift/cluster-node-tuning-operator/pull/1019#pullrequestreview-2073382408), and doing so would simplify the review and finalization of #1019 itself.

### CODE REVIEW

**Review by jmencak — COMMENTED** [review]
Thank you for the changes. This is a definite improvement over the current code. I'm only not sure why one of the log messages was dropped.  Is it no longer useful?

File: pkg/operator/controller.go [hunk 1]
```diff
@@ -685,55 +679,54 @@ func (c *Controller) syncProfile(tuned *tunedv1.Tuned, nodeName string) error {
...
+	profile.Spec.Profile = computed.AllProfiles
 	profile.Status.Conditions = tunedpkg.InitializeStatusConditions()
 
-	klog.V(2).Infof("syncProfile(): updating Profile %s [%s]", profile.Name, tunedProfileName)
+	klog.V(2).Infof("syncProfile(): updating Profile %s [%s]", profile.Name, computed.TunedProfileName)
 	_, err = c.clients.Tuned.TunedV1().Profiles(ntoconfig.WatchNamespace()).Update(context.TODO(), profile, metav1.UpdateOptions{})
 	if err != nil {
 		return fmt.Errorf("failed to update Profile %s: %v", profile.Name, err)
 	}
-	klog.Infof("updated profile %s [%s]", profile.Name, tunedProfileName)
```
[the remark this answers is not in the document]
Comment (ffromani) [reply pkg/operator/controller.go]: no good reason, bad rebase. Fixed, thanks for pointing this out!

**Comment by ffromani** [comment]
> Thank you for the changes. This is a definite improvement over the current code. I'm only not sure why one of the log messages was dropped. Is it no longer useful?

No, I just fumbled up the rebase :) fixed

**Comment by ffromani** [comment]
> > Thank you for the changes. This is a definite improvement over the current code. I'm only not sure why one of the log messages was dropped. Is it no longer useful?
> 
> No, I just fumbled up the rebase :) fixed

aaaaand I fumbled again. It should be OK now.

**Comment by jmencak** [comment]
/lgtm
I believe we can use the same OCPBUG for this as #1019.

**Comment by ffromani** [comment]
> /lgtm I believe we can use the same OCPBUG for this as #1019.

thanks, done

**Pull request merged**